pkg/cloud/vsphere/actuators/machine/actuator.go

```diff
@@ -254,3 +256,95 @@ func (a *Actuator) reconcileReadyState(ctx *context.MachineContext) error {
 	}
 	return nil
 }
+
+// isFirstControlPlaneMachine returns a flag indicating whether or not this
+// machine is the first control plane machine. If it's not, then a requeue
+// error is returned.
+func (a *Actuator) isFirstControlPlaneMachine(ctx *context.MachineContext) (bool, error) {
+
+	// Does the cluster object have the ready annotation label?
+	var controlPlaneReady bool
+	if ok, _ := strconv.ParseBool(ctx.Cluster.Annotations[constants.ReadyAnnotationLabel]); ok {
+		controlPlaneReady = true
+	}
+
+	// If the cluster does not have the ready annotation label, check to see
+	// if any of the control plane machines have the annotation set.
+	if !controlPlaneReady {
+		// Iterate all the control plane machines to check if any of them have the
+		// ready annotation.
+		controlPlaneMachines, err := ctx.GetControlPlaneMachines()
+		if err != nil {
+
+			// An unexpected error occurred. Do not requeue, abort.
+			return false, errors.Wrapf(err, "error getting control plane machines for %q", ctx)
+		}
+		for _, m := range controlPlaneMachines {
+			if ok, _ := strconv.ParseBool(m.Annotations[constants.ReadyAnnotationLabel]); ok {
+				controlPlaneReady = true
+				break
+			}
+		}
+	}
+
+	// If the control plane is ready then there's no need to go any further.
+	if controlPlaneReady {
+		return false, nil
+	}
+
+	// If the control plane is *not* ready, then can this machine initialize
+	// the control plane? First, does it even have the control plane role?
+	// If not then all this worker node can do is requeue until the control
+	// plane is ready.
+	if !ctx.HasControlPlaneRole() {
+		return false, errors.Wrapf(
+			&clustererr.RequeueAfterError{RequeueAfter: constants.DefaultRequeue},
+			"machine is worker node but control plane is not ready for %q", ctx)
+	}
+
+	// The machine has the control plane role, but can it acquire the config
+	// map that gates access to initializing the control plane?
+	blockOwnerDeletion := true
```

Comment:
Your blocking owner deletion so you need to delete this configmap as part of cluster delete.

Reply:
But also, do we need to keep this config map around after the first node is done initializing?

Reply:
Huh. I copied this bit from @ncdc's PR. Andy, is there a reason we're blocking owner deletion?

Reply:
> But also, do we need to keep this config map around after the first node is done initializing?

Hi @andrewsykim,

To that, perhaps, perhaps not. Please see this [Slack conversation](https://kubernetes.slack.com/archives/C8TSNPY4T/p1561736784496000), which actually started [here](https://kubernetes.slack.com/archives/C8TSNPY4T/p1561736365487900) before being moved into the thread.

Reply:
I think it *does* make sense for the Cluster.Reconcile function to delete the config map once the control plane is online. Since the function that provides the locking checks both the cluster and machines for the ready annotation, this will prevent a race condition with the deletion of the config map. Good call @andrewsykim

Reply:
Actually, while I resolved the cleanup, I will leave this conversation open util @ncdc chimes in about the blockOwnerDeletion bit.

Reply:
As long as the configmap is deleted as part of the cluster deletion, either way is fine. I don't recall specifically why I chose to block - I was probably copying another ownerRef setup from CAPI/CAPA.